src/kafka/protocol/api_versions_response.hh

```diff
@@ -28,12 +28,21 @@ namespace seastar {
 
 namespace kafka {
 
+struct unsupported_version_exception : public std::exception {
+    [[nodiscard]] const char *what() const noexcept override {
+        return "The client is not compatible with broker.";
+    }
+};
+
 class api_versions_response_key {
 public:
     kafka_int16_t _api_key;
     kafka_int16_t _min_version;
     kafka_int16_t _max_version;
 
+    bool operator<(const api_versions_response_key& other) const noexcept;
+    bool operator<(int16_t api_key) const noexcept;
```

Comment:
I guess if we already use boost we could use boost_operators to get other comparison ops for free. What do you think?

Reply:
This `operator<` is only defined for the purposes of `std::sort` and `std::lower_bound`, so other operators are not needed now.